core/jobs/batch_jobs/test_gcs_io_job_test.py

```diff
@@ -0,0 +1,120 @@
+
+storage_services = models.Registry.import_storage_services()
+app_identity_services = models.Registry.import_app_identity_services()
+
+
+class TestGCSIoWriteJobTests(job_test_utils.JobTestBase):
+
+    JOB_CLASS = test_gcs_io_job.TestGCSIoWriteJob
+
+    def test_to_fetch_exp_filename(self) -> None:
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(stdout='TOTAL FILES WRITTEN SUCCESS: 2')
+        ])
+
+
+def write_files_to_gcs() -> List[str]:
+    """Write dummy files to GCS."""
+    bucket = app_identity_services.get_gcs_resource_bucket_name()
+    fileobjects = [
+        {
+            'filepath': 'dummy_folder/dummy_subfolder/dummy_file_1',
+            'data': b'testing_1'
+        },
+        {
+            'filepath': 'dummy_folder/dummy_subfolder/dummy_file_2',
+            'data': b'testing_2'
+        }
+    ]
+
+    filepath_0 = fileobjects[0]['filepath']
+    assert isinstance(filepath_0, str)
+    data_0 = fileobjects[0]['data']
+    assert isinstance(data_0, bytes)
+    filepath_1 = fileobjects[1]['filepath']
+    assert isinstance(filepath_1, str)
+    data_1 = fileobjects[1]['data']
+    assert isinstance(data_1, bytes)
+
+    storage_services.commit(
+        bucket, filepath_0, data_0, 'application/octet-stream')
+    storage_services.commit(
+        bucket, filepath_1, data_1, 'application/octet-stream')
+
+    return [
+        'dummy_folder/dummy_subfolder/dummy_file_1',
+        'dummy_folder/dummy_subfolder/dummy_file_2']
+
+
+class TestGCSIoReadJobTests(job_test_utils.JobTestBase):
+
+    JOB_CLASS = test_gcs_io_job.TestGCSIoReadJob
+
+    def test_to_fetch_filename(self) -> None:
+        write_files_to_gcs()
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(
+                stdout='TOTAL FILES FETCHED SUCCESS: 2'),
+            job_run_result.JobRunResult(
+                stdout='The data for file \'dummy_folder/dummy_subfolder/'
+                'dummy_file_1\' is "b\'testing_1\'"'),
+            job_run_result.JobRunResult(
+                stdout='The data for file \'dummy_folder/dummy_subfolder/'
+                'dummy_file_2\' is "b\'testing_2\'"')
+        ])
+
+
+class TestGCSIoGetFilesJobTests(job_test_utils.JobTestBase):
+
+    JOB_CLASS = test_gcs_io_job.TestGcsIoGetFilesJob
+
+    def test_to_fetch_filename(self) -> None:
+        blobs = write_files_to_gcs()
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(
+                stdout='TOTAL PREFIXES FETCHED SUCCESS: 1'),
+            job_run_result.JobRunResult(
+                stdout='The data is %s' % (blobs))
+        ])
+
+
+class TestGCSIoDeleteJobTests(job_test_utils.JobTestBase):
+
+    JOB_CLASS = test_gcs_io_job.TestGcsIoDeleteJob
+
+    def test_to_delete_filenames(self) -> None:
+        write_files_to_gcs()
+        self.assert_job_output_is([
+            job_run_result.JobRunResult(stdout='TOTAL FILES DELETED SUCCESS: 2')
+        ])
```

Comment:
Add more tests and test that the file was deleted properly.

Reply:
I have updated the existing test, where I check if the file exists or not.